packages/babel-preset-expo/plugins/disable-unambiguous-requires.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * Disable unambiguous module ID requires from React Native:
+ * https://github.com/facebook/react-native/commit/06b5bda34923b68ba5141e78c36ccbdc5f4bcff1
+ *
+ * Without this operation, the following error will be thrown when bundling with Webpack:
+ * `Critical dependency: require function is used in a way in which dependencies cannot be statically extracted`
+ *
+ * - react-native/Libraries/Performance/Systrace.js 124:2-9
+ * - react-native/Libraries/Core/setUpReactRefresh.js 30:2-9
+ */
+module.exports = () => ({
+  visitor: {
+    AssignmentExpression(path) {
+      if (!isValidRequire(path) && (isCastedRequire(path) || isChainedRequire(path))) {
+        path.remove();
+      }
+    },
+  },
+});
+
+/**
+ * Is a require statement formatted properly like: `= require('...')`
+ *
+ * @param {*} path
+ */
+function isValidRequire(path) {
+  return path.node.operator !== '=' || path.node.left.type !== 'MemberExpression';
+}
```

Comment:
Does this ever make a difference. It seems like the casts and property assignment tests wouldn't overlap with this.